### DISCUSSION TITLE
Handbrake cannot find Libraries for reading DVDs and Blu-Rays

### DISCUSSION THREAD

**Discussion by dyseptical** [Homebrew/discussions#5681] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 70672606c6ac07a5e8f75abeda7b8736e8285dbe
Last commit: 4 days ago
Core tap JSON: 17 Oct 18:18 UTC
Core cask tap JSON: 17 Oct 18:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit arm_ibiza
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.0.0.0.1.1724870825
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi there! I ran into an issue with the HandBrake Cask and im running out of ideas to try.

Heres the problem:
I installed Handbrake (as Cask), libdvdcss and libaacs for reading DVDs and Blu-Rays in HandBrake.
It unfortunately looks like Handbrake isn't looking for Libraries in /opt/homebrew/opt/..., but rather only in these Paths:
/usr/local/lib/ and /System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/

For reading DVDs, copying the libraries from /opt/homebrew/Cellar/libdvdcss/1.4.3/lib/ to /usr/local/lib was enough.
I just cannot get Blu-Rays working, though. 

Copying the libs of libaacs and libgcrypt over to /usr/local/lib got me a bit further. HandBrake finds libaacs just fine, but libaacs is (i guess?) pointing to /opt/homebrew/opt/libgcrypt/lib/libgcrypt.20.dylib for libgcrypt. HandBrake tries to load this file, which results in this error: "file system sandbox blocked open()".

Heres the error log of HandBrake WITHOUT changing anything after installing with HomeBrew:

> [20:24:48] macgui: ScanCore trying to open a physical disc at: /Volumes/MYBDTITLE
> [20:24:48] macgui: dlopen error: dlopen(/usr/local/lib/libaacs.dylib, 0x0001): tried: '/usr/local/lib/libaacs.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/lib/libaacs.dylib' (no such file), '/usr/local/lib/libaacs.dylib' (no such file)
> [20:24:48] macgui: ScanCore, library not found for decrypting physical disc
> [20:24:50] macgui: User overrode copy-protection warning - trying to open physical dvd without decryption
> [20:24:50] macgui: ScanCore scanning titles with a duration of 60 seconds or more
> [20:24:50] CPU: Apple M3
> [20:24:50]  - logical processor count: 8
> [20:24:50] hb_scan: path=/Volumes/MYBDTITLE, title_index=0
> aacs.c:121: No usable AACS libraries found!
> bdj.c:294: Unable to read path from /usr/libexec/java_home
> bdj.c:795: BD-J check: Failed to load JVM library
> bdj.c:294: Unable to read path from /usr/libexec/java_home
> bdj.c:795: BD-J check: Failed to load JVM library

And here is the error log AFTER moving libaacs and libgcrypt libraries to /usr/local/lib:

> [20:30:05] macgui: ScanCore trying to open a physical disc at: /Volumes/MYBDTITLE
> [20:30:05] macgui: dlopen error: dlopen(/usr/local/lib/libaacs.dylib, 0x0001): Library not loaded: /opt/homebrew/opt/libgcrypt/lib/libgcrypt.20.dylib
>   Referenced from: <122FFDC1-349D-34BE-AC14-1A63CE412EAA> /usr/local/lib/libaacs.dylib
>   Reason: tried: '/opt/homebrew/opt/libgcrypt/lib/libgcrypt.20.dylib' (file system sandbox blocked open()), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libgcrypt/lib/libgcrypt.20.dylib' (no such file), '/opt/homebrew/opt/libgcrypt/lib/libgcrypt.20.dylib' (file system sandbox blocked open())
> [20:30:05] macgui: ScanCore, library not found for decrypting physical disc
> [20:30:06] macgui: User overrode copy-protection warning - trying to open physical dvd without decryption
> [20:30:06] macgui: ScanCore scanning titles with a duration of 60 seconds or more
> [20:30:06] CPU: Apple M3
> [20:30:06]  - logical processor count: 8
> [20:30:06] hb_scan: path=/Volumes/MYBDTITLE, title_index=0
> aacs.c:121: No usable AACS libraries found!
> bdj.c:294: Unable to read path from /usr/libexec/java_home
> bdj.c:795: BD-J check: Failed to load JVM library
> bdj.c:294: Unable to read path from /usr/libexec/java_home
> bdj.c:795: BD-J check: Failed to load JVM library

Can i rebuild the libaacs library without it pointing to the /opt/homebrew/... folder? Or allow HandBrake access to that location?
Maybe theres another solution i don't know about....

Thank you very much in advance!

### Relevant casks

handbrake, libaacs and dependencies, libdvdcss

**Comment by p-linnane** [comment]
Casks are just pointers to upstream's distributed binaries. Any issues should be reported upstream for them to fix in a new release.
